app/test/unit/git/stash-test.ts

```diff
@@ -231,22 +231,67 @@ describe('git/stash', () => {
+      it('restores changes back to the working directory', async () => {
+        await generateTestStashEntry(repository, 'master', true)
+        const entries = await getDesktopStashEntries(repository)
+        expect(entries.length).toBe(1)
+
+        let status = await getStatusOrThrow(repository)
+        let files = status.workingDirectory.files
+        expect(files).toHaveLength(0)
+
+        const entryToApply = entries[0]
+        await popStashEntry(repository, entryToApply.stashSha)
+
+        status = await getStatusOrThrow(repository)
+        files = status.workingDirectory.files
+        expect(files).toHaveLength(1)
+      })
+    })
 
-    it('restores changes back to the working directory', async () => {
-      await generateTestStashEntry(repository, 'master', true)
-      const entries = await getDesktopStashEntries(repository)
-      expect(entries.length).toBe(1)
-
-      let status = await getStatusOrThrow(repository)
-      let files = status.workingDirectory.files
-      expect(files).toHaveLength(0)
-
-      const entryToApply = entries[0]
-      await popStashEntry(repository, entryToApply.stashSha)
-
-      status = await getStatusOrThrow(repository)
-      files = status.workingDirectory.files
-      expect(files).toHaveLength(1)
+    describe('when there are (resolvable) conflicts', () => {
+      it('restores changes and drops stash', async () => {
+        await generateTestStashEntry(repository, 'master', true)
+        const entries = await getDesktopStashEntries(repository)
+        expect(entries.length).toBe(1)
+
+        const readme = path.join(repository.path, 'README.md')
+        await FSE.appendFile(readme, Math.random()) // eslint-disable-line insecure-random
+        await GitProcess.exec(
+          ['commit', '-am', 'later commit'],
+          repository.path
+        )
+
+        let status = await getStatusOrThrow(repository)
+        let files = status.workingDirectory.files
+        expect(files).toHaveLength(0)
+
+        const entryToApply = entries[0]
+        await popStashEntry(repository, entryToApply.stashSha)
+
+        status = await getStatusOrThrow(repository)
+        files = status.workingDirectory.files
+        expect(files).toHaveLength(1)
+
+        const entriesAfter = await getDesktopStashEntries(repository)
+        expect(entriesAfter).not.toContain(entryToApply)
+      })
+    })
+    describe('when there are unresolvable conflicts', () => {
+      it('throws an error', async () => {
+        await generateTestStashEntry(repository, 'master', true)
+        const entries = await getDesktopStashEntries(repository)
+        expect(entries.length).toBe(1)
+
+        const readme = path.join(repository.path, 'README.md')
+        await FSE.writeFile(readme, Math.random()) // eslint-disable-line insecure-random
+
+        const entryToApply = entries[0]
+        expect(
+          popStashEntry(repository, entryToApply.stashSha)
+        ).rejects.toThrowError()
```

Comment:
🤔 Didn't we have issues with using `.rejects` like this in the past? 

I pointed @iAmWillShepherd to this pattern in an earlier stashing PR which would be more reliable, despite me feeling sad about its verbosity...

```ts
      try {
        await dropDesktopStashEntry(repository, doesNotExist.stashSha)
      } catch {
        didFail = true
      }

      expect(didFail).toBe(false)
```

Reply:
ugh i totally forgot about that. looks like jest is still trying to figure this out (https://github.com/facebook/jest/issues/8140). 😞 i tried the pattern above and a couple things i found in the jest docs. it looks like the behavior in https://github.com/desktop/desktop/pull/7432/commits/e9abdc153608df3fee28c6fe9ef1eed6ff6661c6 is roughly equivalent if that works for you?

Reply:
> it looks like the behavior in [e9abdc1](https://github.com/desktop/desktop/commit/e9abdc153608df3fee28c6fe9ef1eed6ff6661c6) is roughly equivalent if that works for you?

If we're fine with that, I'm happy to keep it around. Would also be good to track cleaning up any more of these usages that might be lurking around, but that's not a blocker on this PR.

Reply:
> Would also be good to track cleaning up

i can make an issue for that later today :)